### PR TITLE
fix(composer): deleting a folder should move containing items into parent folder

### LINKED ISSUE #5165: [organization] deleting a folder removes all the items in the folder without a way to navigate back to them
## Motivation

One day we will have the ability to discover unlinked content and search and browse dynamically. Until that day, we need undeleted content to remain always visible and navigable from the left sidebar. We support the ability to delete folders; deleting a folder should not hide all of the content inside that folder.

## Finish Line

- When deleting a folder, the content that was inside the deleted folder should be moved to be parallel to the folder that was deleted

### PR DESCRIPTION
**Pull request by jessmartin**
fixes #5165

### CODE REVIEW

File: packages/apps/composer-app/src/playwright/single-player.spec.ts [hunk 1]
```diff
@@ -39,6 +39,30 @@ test.describe('Single-player tests', () => {
...
 
+  test('create folder', async () => {
+    await host.createSpace();
+    await host.createFolder();
+    await waitForExpect(async () => {
+      expect(await host.getFoldersCount()).to.equal(1);
+    });
+  });
+
+  test('delete folder', async () => {
```
Comment (jessmartin) [inline packages/apps/composer-app/src/playwright/single-player.spec.ts:50]: make this a describe block with two tests:
- delete a folder with a doc in it
- delete a folder with another folder inside it

File: packages/apps/composer-app/src/playwright/app-manager.ts [hunk 2]
```diff
@@ -60,6 +60,17 @@ export class AppManager {
     return this.page.getByTestId(`${plugin}.createObject`).last().click();
   }
 
+  async createFolder() {
+    await this.page.getByTestId('spacePlugin.createObject').last().click();
+    return this.page.getByTestId('spacePlugin.createFolder').last().click();
+  }
+
+  async deleteObject(itemNumber: number) {
```
Comment (jessmartin) [inline packages/apps/composer-app/src/playwright/app-manager.ts:68]: maybe make this `deleteFolder` and use the `testId` of `spacePlugin.folder`?

make another one for `deleteObject`?

**Review by jessmartin — COMMENTED** [review]
some review comments for myself

File: packages/apps/composer-app/src/playwright/single-player.spec.ts [hunk 3]
```diff
@@ -39,6 +39,50 @@ test.describe('Single-player tests', () => {
...
+  test('create folder', async () => {
+    await host.createSpace();
+    await host.createFolder();
+    await waitForExpect(async () => {
+      expect(await host.getFoldersCount()).to.equal(1);
+    });
+  });
+
+  test.describe('deleting folders', () => {
+    test.only('moves item out of folder', async () => {
```
Comment (wittjosiah) [inline packages/apps/composer-app/src/playwright/single-player.spec.ts:51]:
```suggestion
    test('moves item out of folder', async () => {
```

File: packages/apps/composer-app/src/playwright/single-player.spec.ts [hunk 4]
```diff
@@ -39,6 +39,50 @@ test.describe('Single-player tests', () => {
...
+      });
+      // delete the containing folder
+      await host.deleteObject(0);
+
+      await waitForExpect(async () => {
+        expect(await host.getObjectsCount()).to.equal(2);
+      });
+    });
+
+    test.only('moves folder with item out of folder', async () => {
```
Comment (wittjosiah) [inline packages/apps/composer-app/src/playwright/single-player.spec.ts:67]:
```suggestion
    test('moves folder with item out of folder', async () => {
```

**Review by wittjosiah — APPROVED** [review]

**Pull request merged**